pkg/sensors/tracing/args_linux.go

```diff
@@ -263,18 +264,43 @@ func getArg(r *bytes.Reader, a argPrinter) api.MsgGenericKprobeArg {
 		arg.Label = a.label
 		return arg
 	case gt.GenericSockaddrType:
-		var address api.MsgGenericKprobeSockaddr
+		var family uint16
+		if err := binary.Read(r, binary.LittleEndian, &family); err != nil {
+			logger.GetLogger().Warn("failed to read sockaddr family", logfields.Error, err)
+			return nil
+		}
+
+		if _, err := r.Seek(6, io.SeekCurrent); err != nil {
+			logger.GetLogger().Warn("failed to skip sockaddr padding", logfields.Error, err)
+			return nil
+		}
+
 		var arg api.MsgGenericKprobeArgSockaddr
+		arg.Family = family
+
+		switch family {
+		case unixsys.AF_INET, unixsys.AF_INET6:
+			var in api.SockaddrIn
+			if err := binary.Read(r, binary.LittleEndian, &in); err != nil {
+				logger.GetLogger().Warn("failed to read sockaddr_in", logfields.Error, err)
+				return nil
+			}
+			arg.SinAddr = network.GetIP(in.SinAddr, family).String()
+			arg.SinPort = uint32(in.SinPort)
+
+		case unixsys.AF_UNIX:
+			var un api.SockaddrUn
+			if err := binary.Read(r, binary.LittleEndian, &un); err != nil {
+				logger.GetLogger().Warn("failed to read sockaddr_un", logfields.Error, err)
+				return nil
+			}
+			arg.Path = string(bytes.Trim(un.SunPath[:], "\x00"))
```

Comment:
This feels a bit brittle. I feel that we could add a new type (gt.GenericSockaddrUnType) and rename the existing gt.GenericSockaddrType to gt.GenericSockaddrInType, then we can just binary.Read() directly into an appropriate matching data structures in the appropriate cases.